proto3-suite.cabal

```diff
@@ -43,6 +48,8 @@ library
     exposed-modules:   Proto3.Suite.DotProto.Generate.Swagger
     build-depends:     swagger2 >=2.1.6 && <2.7
     cpp-options:       -DSWAGGER
+    if flag(swagger-wrapper-format)
+      cpp-options:       -DSWAGGER_WRAPPER_FORMAT
```

Comment:
Instead of using `CPP`, a cleaner approach is to move all of the code into a separate module, which is conditionally included in the `.cabal` file.  In other words:

```
if flag(swagger-wrapper-format)
  hs-source-dirs: swagger
```

The other benefit of doing it this way is that it also lets you conditionally depend on any necessary modules (not that we need that in this particular case)

Reply:
Yeah you're right, in this case it would be a lot cleaner that way.

Reply:
Hmm actually, I'm not so sure. I think I still need `-XCPP` for the conditional `import`. I'm inclined to stick with just `-XCPP`, and not `-XCPP` + a new module. As much as I hate `-XCPP`, in this case I think it's already as simple as it's going to get.

Reply:
I think this is something better addressed in a future, more broad refactoring.

Reply:
You don't need a conditional import if you do it like this:

* Decide on the name for the separate module (e.g. `Proto3.Suite.DotProto.Generate.Instances`)
* Create two source directories: `instances` and `no-instances`
* Under `no-instances`, place a blank module of the desired name
* Under `instances`, place the module with the instances
* Use an `if` expression in the `.cabal` file to determine which source directory to import based on the configure flag
* `Proto3.Suite.DotProto.Generate.Swagger` unconditionally imports the desired module

Reply:
Ah good point. I like that idea. Would you mind if I tackled that in a follow-up PR, so I can apply that technique to _all_ our flags?

Reply:
Alright, that's fine